sdk/storage/storage-queue/src/QueueClient.ts

```diff
@@ -675,7 +675,7 @@ export class QueueClient extends StorageClient {
     }
     super(url, pipeline);
     this._name = this.getQueueNameFromUrl();
-    this.queueContext = new Queue(this.storageClientContext);
+    this.queueContext = new Queue(this.storageClient);
```

Comment:
Earlier with the V4 generator the StorageClient is extended from Client Context. That design is removed in V6. But, in relaity we do not want anything from Client Context. We want only the Client. So, I changed it here.

Reply:
I see StorageClient still extends StorageClientContext but with some private properties that can only be used inside  StorageClient?  There's a `new Queue(this)` call in the constructor, and here we created another `Queue` via `new Queue(this.storageClient)`. It just feel weird.

![image](https://user-images.githubusercontent.com/7583839/107550574-0d6d7900-6b86-11eb-8840-1ca30a2770bc.png)

Reply:
Looking at the changes in ClientContext.ts it looks like we should be able to keep using it directly

Reply:
@joheredi Actually, I have checked this part. Let me explain. The problem does not start with StorageClient and StorageClientContext classes. The issue starts with Queue class.

The Class structure is:

```StorageClient``` extends ```StorageClientContext``` extends ```ServiceClient```. (Same in V4 and V6 generators)

But, for the generated ```Queue``` operation class, in V4 generator it extends from ```StorageClientContext```. In V6, it extends from ```StorageClient``` class, i.e. in V6 it extends from a Sub Class and not a Super Class. 

Having said that, Now if you look at the code change, it will be more clear.

Reply:
> But, for the generated `Queue` operation class, in V4 generator it extends from `StorageClientContext`. In V6, it extends from `StorageClient` class

what is the reason behind the design change?  It looks like a valid scenario to extend from `StorageClientContext` in this example.

Reply:
Fixed it